kubernetes/kubernetes.go

```diff
@@ -245,7 +246,17 @@ func (ka KubeAgentConfig) collectMetrics(ctx context.Context, config KubeAgentCo
 
 	sampleStartTime := time.Now().UTC()
 
-	//create metric sample directory
+	// refresh client token before each collection
+	token, err := getBearerToken(config.BearerTokenPath)
+	if err != nil {
+		log.Warnf("Warning: %s. If cluster version >=1.21 this token must be updated", err)
```

Comment:
Small note: I think we can tweak this log to make it a bit better. The user isn't going to know which token 'this' token is when going through logs so its better to be a bit more descriptive. I think it's also good  to inform them why they'd care about it (i.e. If token doesn't refresh, metrics-agent won't be able to collect data once that token expires). Not a blocker, will let you decide if you want to change now or cut a ticket to tweak logs